### PR TITLE
Add Spanish locale

### PR DESCRIPTION
**Pull request by PwtKCL**
See commits for details.

### CODE REVIEW

File: src/localisation/en.json [hunk 1]
```diff
@@ -97,7 +97,7 @@
         "errorUserDefinedFuncMsg":"User-defined function\nthis function definition contains errors",
         "errorUserDefinedVarMsg":"User-defined variable\none of the assigments for this variable contains errors",
         "EOFError": "This frame cannot be empty.",
-        "GDriveSaveFailed":"Strype coud not saved to Google Drive. Your latest changes may have not been saved, and the sync has been turned off.",
+        "GDriveSaveFailed":"Strype coud not save to Google Drive. Your latest changes may have not been saved, and the sync has been turned off.",
```
Comment (neilccbrown) [inline src/localisation/en.json:100]: One more correction: coud->could.  And actually, two lines up: assigments->assignments.  Maybe I should have a pass through at some point.

File: src/localisation/es.json [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+        "resetProject":"Nuevo proyecto",
+        "resetProjectTooltip":"Restablecer el editor con el proyecto predeterminado de Strype",
+        "prefs":"Preferencias",
+        "lang":"Idioma: ",
+        "version":"versión",
+        "homepage":"Página de inicio de Strype"
+    },
+
+    "autoCompletion": {
+        "myFunctions":"Mis funciones ",
```
Comment (neilccbrown) [inline src/localisation/es.json:183]: We should probably trim the trailing space.

**Review by neilccbrown — CHANGES_REQUESTED** [review]
Did you see that Github Actions tells us warnings that occurred during the tests?  Interesting!  Although actually we may want to disable that one because we do have a console since it's all client-side, and all those console.log look deliberate.

Anyway, just a couple of small typo-like things to fix.

File: src/localisation/en.json (hunk 1 above)
Comment (PwtKCL) [reply src/localisation/en.json:100]: If that's the logs I'm thinking about, yes I was aware of them (but unfortunately I've noticed them only after doing the merge request of the related code). So I'll keep it the related code as is for now but will change it when doing work on bug fixes pretty soon.

File: src/localisation/es.json (hunk 2 above)
Comment (PwtKCL) [reply src/localisation/es.json]: Well spotted

**Comment by PwtKCL** [comment]
Regarding the warnings: I've noticed some warnings during the compilation which I left for the time being, as I need to make some code fixes later anyway so I can do it later. 
But after looking in more details I think you refer to the warnings about "no-console" ? We can discuss it when we meet.

File: src/localisation/en.json (hunk 1 above)
Comment (PwtKCL) [reply src/localisation/en.json]: Done, good point and I've tried to spot other typos in a quick pass with Word (to see the spell checker indicators).

**Pull request merged**